### PR TITLE
Simplify rendering pipeline, reduce gpu memory usage, remove duplicated data

### PR DESCRIPTION
**Pull request by PPakalns**

### CODE REVIEW

**Comment by PPakalns** [comment]
Didn't know that, learned something new :)

**Comment by hecrj** [comment]
I tested these changes together with #90, #91, and #92 in a relatively big app and I see a noticeable speedup on my M1 (from 2ms frame time to 1.7ms). I guess it's expected since the `prepare` loop does quite less. GPU usage seems unaffected.

**Review by hecrj — CHANGES_REQUESTED** [review]

File: src/text_render.rs [hunk 1]
```diff
@@ -2,21 +2,19 @@ use crate::{
...
 
 /// A text renderer that uses cached glyphs to render text into an existing render pass.
 pub struct TextRenderer {
     vertex_buffer: Buffer,
     vertex_buffer_size: u64,
-    index_buffer: Buffer,
-    index_buffer_size: u64,
-    vertices_to_render: u32,
     screen_resolution: Resolution,
+    gylphs_to_render: u32,
```
[the remark this answers is not in the document]
Comment (PPakalns) [reply src/text_render.rs]: Fixed

**Comment by grovesNL** [comment]
I just rebased this onto `main`. Let's go ahead with this for the CPU time improvements.

It's possible it could regress GPU time some specific older hardware that doesn't handle small instances well, but most modern hardware should be ok with this based on what I can tell. e.g., it's confirmed that AMD has optimizations for small instances now instead of doing the slow thing that used to happen.

If we see a noticeable performance problem with any hardware then we can always special case certain vendors and expand the vertices out in those cases.

**Pull request merged**